test/ginkgo-ext/scopes.go

```diff
@@ -344,14 +374,11 @@ func RunAfterEach(cs *scope) {
 	}
 
 	// Only run afterAll when all the counters are 0 and all afterEach are executed
```

Comment:
This is the only part of the code the suggests anything about the bounds of this counter, and based on this I think this counter is definitely assumed to go from 0 to N (when N>0), and back to 0, and any negative values are therefore unexpected. It's possible that flakes where cause by this also, as this counter is just completely off.

Reply:
I've double-checked with @eloycoto and this is defo unexpected.

Reply:
Looks like the negative count was due to my changes here, as `+=` is being used, the initial `-1` gets added up over and over. The previous version was setting children scope counters  in inexplicit way, which is exactly the line that's been dropped in my changes.